e2e/key/key.go

```diff
@@ -0,0 +1,235 @@
+	tests := []struct {
+		name       string
+		exportPath string
+		consoleOps []e2e.SingularityConsoleOp
+	}{
+		{
+			name:       "import public binary",
+			exportPath: c.publicExportPath,
+			consoleOps: []e2e.SingularityConsoleOp{},
+		},
+		{
+			name:       "import private binary",
+			exportPath: c.privateExportPath,
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("e2etests"),
+				e2e.ConsoleSendLine("e2etests"),
+				e2e.ConsoleSendLine("e2etests"),
+			},
+		},
+		{
+			name:       "import public ascii",
+			exportPath: c.publicExportASCIIPath,
+			consoleOps: []e2e.SingularityConsoleOp{},
+		},
+		{
+			name:       "import private ascii",
+			exportPath: c.privateExportASCIIPath,
+			consoleOps: []e2e.SingularityConsoleOp{
+				e2e.ConsoleSendLine("e2etests"),
+				e2e.ConsoleSendLine("e2etests"),
+				e2e.ConsoleSendLine("e2etests"),
+			},
+		},
+	}
+
+	prepCmd := func(exportPath string) []string {
+		return []string{"import", exportPath}
+	}
+
+	for _, tt := range tests {
+		c.singularityResetKeyring(t) // Remove the tmp keyring before each import
+		c.env.RunSingularity(
+			t,
+			e2e.WithPrivileges(false),
+			e2e.WithCommand("key"),
+			e2e.WithArgs(prepCmd(tt.exportPath)...),
+			e2e.ConsoleRun(tt.consoleOps...),
+			e2e.ExpectExit(0),
+		)
+	}
+	//t.Run("singularityKeyExport", c.singularityKeyExport)
+}
+
+func (c *ctx) singularityResetKeyring(t *testing.T) {
+	// TODO: run this as non-root
+	if err := os.RemoveAll(c.keyRing); err != nil {
+		t.Fatalf("unable to remove tmp keyring directory: %s", err)
+	}
+}
+
+// RunE2ETests is the main func to trigger the test suite
+func RunE2ETests(env e2e.TestEnv) func(*testing.T) {
+	c := &ctx{
+		env:                    env,
+		publicExportPath:       filepath.Join(env.TestDir, "public_key.asc"),
+		publicExportASCIIPath:  filepath.Join(env.TestDir, "public_ascii_key.asc"),
+		privateExportPath:      filepath.Join(env.TestDir, "private_key.asc"),
+		privateExportASCIIPath: filepath.Join(env.TestDir, "private_ascii_key.asc"),
+		keyRing:                filepath.Join(env.TestDir, "sypgp-test-keyring"),
+	}
+
+	if err := os.Setenv("SINGULARITY_SYPGPDIR", c.keyRing); err != nil {
+		panic(fmt.Sprintf("unable to set keyring: %s", err))
+	}
+
+	return func(t *testing.T) {
+		//t.Run("singularityKeyNewpairAndExport", c.singularityKeyNewpair)
+		//t.Run("singularityKeyImportAndExport", c.singularityKeyImport)
+		t.Run("singularityKeyNewpair", c.singularityKeyNewpair) // Generate a newpair (required for the other tests)
+		t.Run("singularityKeyExport", c.singularityKeyExport)   // Export the newpair
+		t.Run("singularityKeyImport", c.singularityKeyImport)   // Import the newpair (this will delete the old, tmp keyring before importing)
+		t.Run("singularityKeyExport", c.singularityKeyExport)   // Re-export them, again (this will catch any issues if Singularity cant import correctly)
+		t.Run("singularityKeyImport", c.singularityKeyImport)   // Finally, Re-import them
```

Comment:
please do not rely on a specific order for tests. If you need for things to happen in a particular order, make them part of the same test.